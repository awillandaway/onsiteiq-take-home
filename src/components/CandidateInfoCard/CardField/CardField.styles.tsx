import styled from 'styled-components';

export const CardFieldWrapper = styled.div`
  display: flex;
  padding: 4px;
  &:not(:last-of-type) {
    border-bottom: 1px solid #282828;
  }
`;

export const CardFieldLabel = styled.div`
  flex: 1;
  font-weight: 500;
`;

export const CardFieldContent = styled.div`
  flex: 1;
`;
