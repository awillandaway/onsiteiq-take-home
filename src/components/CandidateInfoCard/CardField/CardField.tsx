import type { ReactNode } from 'react';
import { CardFieldContent, CardFieldLabel, CardFieldWrapper } from './CardField.styles';

interface CardFieldProps {
  label: string;
  children: ReactNode;
}

export const CardField = ({ label, children }: CardFieldProps) => (
  <CardFieldWrapper>
    <CardFieldLabel>{label}</CardFieldLabel>
    <CardFieldContent>{children}</CardFieldContent>
  </CardFieldWrapper>
);
