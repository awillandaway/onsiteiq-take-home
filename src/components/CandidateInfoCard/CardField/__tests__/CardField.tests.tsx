import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { CardField } from '../CardField';

describe('CardField', () => {
  it('Renders correctly', () => {
    const cardField = renderer.create(<CardField label="Test label">Test content</CardField>).toJSON();
    expect(cardField).toMatchSnapshot();
  });

  it('Renders label with content', () => {
    const { getByText } = render(<CardField label="Test label">Test content</CardField>);

    expect(getByText('Test label')).toBeVisible();
    expect(getByText('Test content')).toBeVisible();
  });
});
