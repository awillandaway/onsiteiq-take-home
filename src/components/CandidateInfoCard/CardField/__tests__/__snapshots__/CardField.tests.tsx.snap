// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardField Renders correctly 1`] = `
<div
  className="sc-bcXHqe ktVuSG"
>
  <div
    className="sc-gswNZR jDfkgI"
  >
    Test label
  </div>
  <div
    className="sc-dkrFOg kGVctq"
  >
    Test content
  </div>
</div>
`;
